### PR TITLE
MINOR: Fix ConsumerBounceTest to wait for consumer close

### PR DESCRIPTION
**Pull request by rajinisivaram**

### CODE REVIEW

**Comment by rajinisivaram** [comment]
@ijuma Can you take a look (and merge if it is ok), please? Thank you!

This is a fix for the failure in https://builds.apache.org/job/kafka-pr-jdk8-scala2.12/5268/
```
java.util.ConcurrentModificationException: KafkaConsumer is not safe for multi-threaded access
	at org.apache.kafka.clients.consumer.KafkaConsumer.acquire(KafkaConsumer.java:1638)
	at org.apache.kafka.clients.consumer.KafkaConsumer.close(KafkaConsumer.java:1536)
	at org.apache.kafka.clients.consumer.KafkaConsumer.close(KafkaConsumer.java:1514)
	at kafka.api.IntegrationTestHarness.$anonfun$tearDown$2(IntegrationTestHarness.scala:99)
	at kafka.api.IntegrationTestHarness.$anonfun$tearDown$2$adapted(IntegrationTestHarness.scala:99)
```

File: core/src/test/scala/integration/kafka/api/ConsumerBounceTest.scala [hunk 1]
```diff
@@ -333,7 +333,8 @@ class ConsumerBounceTest extends IntegrationTestHarness with Logging {
     val rebalanceFuture = createConsumerToRebalance()
 
     // consumer1 should leave group and close immediately even though rebalance is in progress
-    submitCloseAndValidate(consumer1, Long.MaxValue, None, Some(gracefulCloseTimeMs))
+    val future1 = submitCloseAndValidate(consumer1, Long.MaxValue, None, Some(gracefulCloseTimeMs))
+    future1.get
```
Comment (ijuma) [inline core/src/test/scala/integration/kafka/api/ConsumerBounceTest.scala:337]: Is this needed? It seems like `waitForRebalance` would do the job, no?

File: core/src/test/scala/integration/kafka/api/ConsumerBounceTest.scala [hunk 2]
```diff
@@ -343,7 +344,8 @@ class ConsumerBounceTest extends IntegrationTestHarness with Logging {
     servers.foreach(server => killBroker(server.config.brokerId))
 
     // consumer2 should close immediately without LeaveGroup request since there are no brokers available
-    submitCloseAndValidate(consumer2, Long.MaxValue, None, Some(0))
+    val future2 = submitCloseAndValidate(consumer2, Long.MaxValue, None, Some(0))
+    future2.get
```
Comment (ijuma) [inline core/src/test/scala/integration/kafka/api/ConsumerBounceTest.scala:348]: Maybe we don't need a variable and we can just call `get` on the result?

File: core/src/test/scala/integration/kafka/api/IntegrationTestHarness.scala [hunk 3]
```diff
@@ -95,9 +95,12 @@ abstract class IntegrationTestHarness extends KafkaServerTestHarness {
...
   override def tearDown() {
-    producers.foreach(_.close())
-    consumers.foreach(_.close())
-    super.tearDown()
+    try {
+      producers.foreach(_.close())
+      consumers.foreach(_.close())
+    } finally {
+      super.tearDown()
+    }
```
Comment (ijuma) [inline core/src/test/scala/integration/kafka/api/IntegrationTestHarness.scala:103]: This is a bit tricky. Maybe we should just leave as it was. The issue is that `close` throwing an exception is a bug and if we do the try/finally, we may obscure the original exception if `super.tearDown` itself throws an exception.

**Review by ijuma — COMMENTED** [review]
Thanks for the PR. I think the actual issue here is that we don't wait for the executor's tasks to terminate in the tearDown before we try to close the consumers. If we do that, then we should not have the issue reported here.

One of the `get` calls you added still seems worthwhile. The other one, I am less sure about.

**Comment by rajinisivaram** [comment]
Closing this PR since the test was updated in commit 031da88.

**Pull request closed without merging**